src/NuGet.Services.EndToEnd/Support/Clients/V3IndexClient.cs

```diff
@@ -24,39 +24,57 @@ public V3IndexClient(SimpleHttpClient httpClient, TestSettings testSettings)
             _testSettings = testSettings;
         }
 
-        public async Task<IReadOnlyList<string>> GetSearchBaseUrlsAsync(ITestOutputHelper logger)
+        public async Task<IReadOnlyList<string>> GetSearchGalleryQueryServiceUrlsAsync(ITestOutputHelper logger)
         {
             var v3Index = await GetV3IndexAsync(logger);
-            return GetResourceUrls(v3Index, t => t.Type.StartsWith("SearchGalleryQueryService/"));
+            return GetResourceUrls(v3Index, "SearchGalleryQueryService");
+        }
+
+        public async Task<IReadOnlyList<string>> GetSearchQueryServiceUrlsAsync(ITestOutputHelper logger)
+        {
+            var v3Index = await GetV3IndexAsync(logger);
+            return GetResourceUrls(v3Index, "SearchQueryService");
+        }
+
+        public async Task<IReadOnlyList<string>> GetSearchAutocompleteServiceUrlsAsync(ITestOutputHelper logger)
+        {
+            var v3Index = await GetV3IndexAsync(logger);
+            return GetResourceUrls(v3Index, "SearchAutocompleteService");
         }
 
         public async Task<IReadOnlyList<string>> GetFlatContainerBaseUrlsAsync(ITestOutputHelper logger)
         {
             var v3Index = await GetV3IndexAsync(logger);
-            return GetResourceUrls(v3Index, t => t.Type.StartsWith("PackageBaseAddress/"));
+            return GetResourceUrls(v3Index, "PackageBaseAddress");
         }
 
         public async Task<IReadOnlyList<string>> GetRegistrationBaseUrlsAsync(ITestOutputHelper logger)
         {
             var v3Index = await GetV3IndexAsync(logger);
-            return GetResourceUrls(v3Index, t => t.Type.StartsWith("RegistrationsBaseUrl/"));
+            return GetResourceUrls(v3Index, "RegistrationsBaseUrl");
         }
 
         public async Task<IReadOnlyList<string>> GetSemVer2RegistrationBaseUrlsAsync(ITestOutputHelper logger)
         {
             var v3Index = await GetV3IndexAsync(logger);
             return GetResourceUrls(
                 v3Index,
-                t => t.Type == "RegistrationsBaseUrl/Versioned" &&
-                     t.ClientVersion != null &&
+                "RegistrationsBaseUrl/Versioned",
+                t => t.ClientVersion != null &&
                      NuGetVersion.Parse(t.ClientVersion) >= Version430Alpha);
         }
 
-        private static List<string> GetResourceUrls(V3Index v3Index, Func<Resource, bool> isMatch)
+        private static List<string> GetResourceUrls(V3Index v3Index, string resourceType, Func<Resource, bool> isMatch = null)
```

Comment:
Should we just have separate overloads one for `.StartsWith($"{resourceType}/")` and another one for `isMatch`? This looks a cramming everything into the single method just to be able to call the same method in `GetSemVer2RegistrationBaseUrlsAsync`.

Reply:
Yeah, you're totally right. I'll do that.

Reply:
Done.